invenio_vocabularies/contrib/names/services.py

```diff
@@ -8,8 +8,49 @@
 
 """Names services."""
 
+from elasticsearch_dsl.query import Q
+from invenio_pidstore.errors import PIDDoesNotExistError
+from invenio_records_resources.config import lt_es7
+
 from .names import record_type
 
 NamesServiceConfig = record_type.service_config_cls
 
-NamesService = record_type.service_cls
+
+class NamesService(record_type.service_cls):
+    """Name service."""
+
+    def resolve(self, identity, id_, id_type):
```

Comment:
Shouldn't this be using the DB instead of ES, to resolve to a single Name record? Each "formal" identifier we have in the metadata could be a PID that gets resolved.

Reply:
IRL: A decission needs to be taken on making a `PIDSField` (systemfield). This would allow to do DB resolution, instead of going through ES.